operators/audio/audio_decoder.hpp

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
can some of the implementation details be put into a .cc file?

Reply:
The big advantage of using header file is we don't need to do if/else in CMake file, plus function declaration and implementation are unified, i.e. the function prototype can be changed just in one place.

Reply:
> we don't need to do if/else in CMake file

not quite following what this if/else in CMake is - wouldn't it just be adding another file?

a disadvantage of doing this is adding unnecessary stuff for whoever includes this file. for example, macro definitions like DR_FLAC_IMPLEMENTATION and inclusion of external library headers. IMO, the encapsulation is worth splitting things into both a header and implementation file.

Reply:
If we add more audio custom ops in the future, and all of them implemented in headers file, we don't need to glob cc files according to different pattern, like what we did for tokenizer files and they looks a little messy.

.hpp extension file in the projects is not indented to be included in the multiple places, while .h files are common which are designed to be included by other header files.

Reply:
> .hpp extension file in the projects is not indented to be included in the multiple places, while .h files are common which are designed to be included by other header files.

I was not aware of this convention. is it documented anywhere?